test/firrtl/connect.mlir

```diff
@@ -0,0 +1,408 @@
+// RUN: circt-opt %s -split-input-file -verify-diagnostics
+
+/// Analog types cannot be connected and must be attached.
```

Comment:
nice testcase!

Reply:
I agree! Thanks for painstakingly enumerating all these conditions. :+1: